File: server/packages/kb-biz/kb_biz/schemas/knowledge.py
from __future__ import annotations

from datetime import datetime
from typing import Optional

from pydantic import BaseModel


class DocumentResponse(BaseModel):
    id: str
    dept_id: str
    title: str
    file_name: str
    file_type: str
    file_size: int
    md5: Optional[str] = None
    visibility: str = "dept"
    status: str
    error_message: Optional[str] = None
    file_path: str = ""
    chunk_count: int = 0
    project_id: Optional[str] = None
    uploaded_by: Optional[str] = None
    created_at: datetime
    updated_at: datetime


class UploadResponse(BaseModel):
    id: str
    file_name: str
    file_type: str
    file_size: int
    status: str


class ReindexResponse(BaseModel):
    id: str
    status: str


class VisibilityUpdate(BaseModel):
    visibility: str

File: frontend/src/types/knowledge.ts
export interface Document {
  id: string
  dept_id: string
  title: string
  file_name: string
  file_type: string
  file_size: number
  md5: string | null
  visibility: 'private' | 'dept' | 'public'
  status: 'pending' | 'parsing' | 'chunking' | 'indexing' | 'ready' | 'failed'
  error_message: string | null
  file_path: string
  chunk_count: number
  uploaded_by: string | null
  created_at: string
  updated_at: string
}

export interface UploadResult {
  id: string
  file_name: string
  file_type: string
  file_size: number
  status: string
}

export interface PaginationMeta {
  total: number
  page: number
  page_size: number
}
